zerver/lib/actions.py

```diff
@@ -132,14 +132,26 @@ def active_user_ids(realm):
     # type: (Realm) -> List[int]
     return [userdict['id'] for userdict in get_active_user_dicts_in_realm(realm)]
 
-def stream_user_ids(stream):
-    # type: (Stream) -> List[int]
-    subscriptions = Subscription.objects.filter(recipient__type=Recipient.STREAM,
-                                                recipient__type_id=stream.id)
-    if stream.invite_only:
-        subscriptions = subscriptions.filter(active=True)
+def can_access_stream_user_ids(stream):
+    # type: (Stream) -> Set[int]
+
+    # return user ids of users who can access the attributes of
+    # a stream, such as its name/description
+    if stream.is_public():
+        return set(active_user_ids(stream.realm))
+    else:
+        return private_stream_user_ids(stream)
+
+def private_stream_user_ids(stream):
+    # type: (Stream) -> Set[int]
+
+    # TODO: Find similar queries elsewhere and de-duplicate this code.
+    subscriptions = Subscription.objects.filter(
+        recipient__type=Recipient.STREAM,
+        recipient__type_id=stream.id,
+        active=True)
 
-    return [sub['user_profile_id'] for sub in subscriptions.values('user_profile_id')]
+    return {sub['user_profile_id'] for sub in subscriptions.values('user_profile_id')}
```

Comment:
I think this won't actually work when passed into `send_event`, because sets are not JSON serializable.